libmscore/cursor.cpp

```diff
@@ -53,22 +53,31 @@ void Cursor::setScore(Score* s)
 
 void Cursor::rewind(int type)
       {
-      if (type == 0) {
-            _segment = 0;
-            Measure* m = _score->firstMeasure();
-            if (m) {
-                  _segment = m->first(Segment::Type::ChordRest);
-                  firstChordRestInTrack();
-                  }
-            }
-      else if (type == 1) {
-            _segment  = _score->selection().startSegment();
-            _track    = _score->selection().staffStart() * VOICES;
-            firstChordRestInTrack();
-            }
-      else if (type == 2) {
-            _segment  = _score->selection().endSegment();
-            _track    = (_score->selection().staffEnd() * VOICES) - 1;  // be sure _track exists
+      switch(type) {
+            case 0:                 // score start of current track
+                  _segment = _score->firstSegment();
```

Comment:
Well, this is true also for the current code, if the current track has no chord/rest at all (which, for voices != 0, is not an unlikely case at all).

In addition, the current code limits the returned range to chord/rests, which I found limitating.